Docs revamp (#22)
upsolve-docs/docs: Build-and-Deploy-Your-First-Dashboard.mdx

Welcome to **Upsolve AI**, your go-to platform for enabling **embedded business intelligence** for your customers. By following this example we will walk you through the essentials of connecting your data, setting up tenancy, building your first dashboard, sharing your insights, and deploying your dashboards in your product.

## Connecting data:

We’ve already set up a **sample dataset** to help you quickly explore and test the platform without needing to connect your own data source. 

To view this connection, please go to the **Connect** tab as seen below:

<Frame>
<img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/data-connection-hero.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/data-connection-hero-dark.png"
  alt="Hero Dark"
/>
</Frame>

> If you would like to connect to a different data source, please go to the [Set up your connection](data-connection/data-connection).

Author: ```suggestion
If you would like to connect to a different data source, please have a look at how to [set up your connection](data-connection/data-connection).
```

  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/select-raw-tables-dark.png"
  alt="Hero Dark"
/>
</Frame>
    <Tip>

    The list of available raw table(s) is generated after the [data connection](/data-connection/data-connection) is set up.

    </Tip>

  </Accordion>

{" "}
<Accordion title="2. Prepare data">
Click on the **+ aggregations** , this will create ```SQL SELECT * FROM "aggregations" ```   
in the SQL editor.

Click on **Preview data** to run the SQL and **Next** once you see the preview data.
     <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/query-builder.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/query-builder-dark.png"
  alt="Hero Dark"
/>
    </Frame>

 <Tip>
    If you are more advanced with SQL, you can add more tables and customise your SQL.
    </Tip>
  </Accordion>

{" "}
<Accordion title="3. Build chart">
  
  For your first chart select a **Grouped bar**, and choose **country** as the x-axis and  **demand** as the y-axis.
  the corresponding x- & y-axis
  

>   For future charts see **BROKEN LINK** [Chart Configuration](/content/charts/area-line-chart-a) for all chart settings. 

Author: ```suggestion
  For future charts see [Chart Configuration](/charts/create-chart) for all chart settings. 
```

Author: or `/content/charts/area-line-chart`

  
  

     <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/chart-builder-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/chart-builder-dark.png"
  alt="Hero Dark"
/>
     </Frame>

</Accordion>

  <Accordion title="4. Configure filters">

>     You can set up filters by clicking on **configure filters** and selecting the relevant column and filter type.

Author: The image here should relate to the text. We should also link to the chart filter page here http://localhost:3002/filters/filters-in-depth#chart-filters

Author: can we give a specific walkthrough? like set up a date filter

Reviewer: Yeah, I need a better dataset, as there are no date filters available

    ![configure-filters](/images/filters/filters-guide-4.png)

  Click **Save and add to dashboard** when you are happy with your chart, this will save the chart. You could also
  create a copy of the chart by clicking **Duplicate**.
    <Tip>
    For a more comprehensive overview, check out our [in-depth filtering guide](/filters/filters-in-depth).
    </Tip>
    
  </Accordion>
</AccordionGroup>

### Configure your dashboard

You are able to customise your dashboard by adding dashboard filters, controlled filters and additional charts. 

To resize charts you can simply drag the chart to the size you would like.

To move a chart simply drag and drop the chart.

#### Add dashboard filters

> To add a dashboard filter please click on **New Filter**

Author: This isn't enough, let's at least link tot he dashboard filter page but better yet walk through concretely 1 filter setuo

   <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/example-dashboard-filter-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/example-dashboard-filter-dark.png"
  alt="Hero Dark"
/>
     </Frame>

#### Add charts

To add additional charts you can go to through the steps of creating a new chart. 

Alternatively if you have already built a chart you can click on **Add Chart** to use on your dashboard.
 <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/add-chart-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/add-chart-dark.png"
  alt="Hero Dark"
/>
     </Frame>

## Share your dashboard

You can share your dashboard via email with anyone in by clicking on the share button and sharing it as follows, if you would like a preview of the dashboard please click on the preview.

> Your team members will be invited to your dashboard by sharing your dashboard they will become a team member.

Author: ```suggestion
Your team members will be invited to your dashboard. By sharing your dashboard they will become a team member and can collaborate on future dashboards with you.
```

<Frame>
<img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/share.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/share-dark.png"
  alt="Hero Dark"
/>
</Frame>

<Note> If the invitee aceepts the invite, they will gain access to ALL dashboards and data in your organization!</Note>

## User Authorization (Tenant Registration)

To set up **tenant registration** please follow the steps outlined in [Backend Setup](deploy-dashboards/backend-setup). 

> Alternatively if you are using our **sample dataset** and do not intend to **go live** with the dashboard this step is optional and you can move to the next step.

Author: I think this should be `sample database` everywhere

> Alternatively if you are using our **sample dataset** and do not intend to **go live** with the dashboard this step is optional and you can move to the next step.

Author: ```suggestion
Alternatively if you are using our **sample database** and do **not intend to go live** with the dashboard this step is optional.
```

Welcome to **Upsolve AI**, your go-to platform for enabling **embedded business intelligence** for your customers. By following this example we will walk you through the essentials of connecting your data, setting up tenancy, building your first dashboard, sharing your insights, and deploying your dashboards in your product.

## Connecting data:

We’ve already set up a **sample database** to help you quickly explore and test the platform without needing to connect your own data source. 

To view this connection, please go to the **Connect** tab as seen below:

<Frame>
<img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/data-connection-hero.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/data-connection-hero-dark.png"
  alt="Hero Dark"
/>
</Frame>

If you would like to connect to a different data source, please have a look at how to [set up your connection](data-connection/data-connection).

## Build your first dashboard

In order to build your first dashboard you should begin by creating your first chart before adding to the dashboard. 

### Build your first chart

> It requires five steps to build your first chart:

Reviewer 2: Rewrite it to (will probably need to change some of the text below when we swap to a new database):

It requires five steps to build your first chart. In the below example, you will build a grouped bar chart using the **aggregations** table.

<AccordionGroup>
  <Accordion title="1. Select raw table(s)">

>     Search and select the **aggregations** table. 

Reviewer 2: Rewrite to:

Search and select the table(s) that contain the relevant data for the chart you want to create, in this case, **aggregations** table.

    <Frame>
<img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/select-raw-tables.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/select-raw-tables-dark.png"
  alt="Hero Dark"
/>
</Frame>
    <Tip>

    The list of available raw table(s) is generated after the [data connection](/data-connection/data-connection) is set up.

    </Tip>

  </Accordion>

{" "}
<Accordion title="2. Prepare data">

> Click on the **+ aggregations** , this will create ```SQL SELECT * FROM "aggregations" ```   

Reviewer 2: Add the below before your text:

Write any SQL query to transform the table(s) you selected to a single table that you will consume in Step 3 for chart building. You could write SQL to join, filter on table(s) and more.

For this example, you simply want to use the whole **aggregations** table for chart building. Click on the **+ aggregations**, this will......

in the SQL editor.

Click on **Preview data** to run the SQL and **Next** once you see the preview data.
     <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/query-builder.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/query-builder-dark.png"
  alt="Hero Dark"
/>
    </Frame>

 <Tip>
    If you are more advanced with SQL, you can add more tables and customise your SQL.
    </Tip>
  </Accordion>

> {" "}

Reviewer 2: for step 3-5, refer to what I did in 1-2, and do the same. We need to include a one liner to explain to user conceptually what thy are doing

<Accordion title="3. Build chart">
  
  For your first chart select a **Grouped bar**, and choose **country** as the x-axis and  **demand** as the y-axis.
  the corresponding x- & y-axis
  
  For future charts see [Chart Configuration](/charts/create-chart) for all chart settings. 
  
  

     <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/chart-builder-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/chart-builder-dark.png"
  alt="Hero Dark"
/>
     </Frame>

</Accordion>

  <Accordion title="4. Add chart filters">
  To add chart filters 
    You can set up filters by clicking on **configure filters** and selecting the relevant column and filter type.

    <Steps>
      <Step title="Select Table ">

>       **1. Raw backing tables:** filter any column in the raw data that is powering this specific chart

Reviewer 2: Lets use consistent wording. Seems like we use table everywhere, if so, replace "raw data" with table. Also strictly speaking, (1) isn't really raw backing tables. They are just selected backing tables

I would say replace it with:
**1. Selected backing tables:** The list of selected backing tables in Step 1 is surfaced here. Filter any column in any selected backing table(s) that is powering this specific chart

> **2. Computed table:** filter any column from the data table that was computed by the SQL written for this chart

Reviewer 2: just use table, and not data table, and say "was computed by the SQL written for this chart in Step 2"

> **3. Variables in the chart SQL:** inject variables into the SQL code that is backing this chart

Reviewer 2: is there doc on variable filtering already in the Filtering section? Let's add a link to it so reader could reference

Reviewer: No, we it as a to-do.

      </Step>

      

      <Step title="Configure Filter ">

**1. Table:** select the table to filter or "Query variables"

**2. Column:** select the column or variable to filter on

**3. Filter:** configure the filter as needed. Options will be provided based on the type of filter selected

      </Step>
    

      
      </Steps>
    ![configure-filters](/images/filters/filters-guide-4.png)

  
    <Tip>
    For a more comprehensive overview, check out our [in-depth filtering guide](/filters/filters-in-depth).
    </Tip>
    
  </Accordion>

    <Accordion title="5. Save chart">

  Click **Save and add to dashboard** when you are happy with your chart, this will save the chart. You could also
  create a copy of the chart by clicking **Duplicate**.

  </Accordion>
</AccordionGroup>

> ### Configure your dashboard

Reviewer 2: would be good to add a screenshot here, that has the chart in the dashboard, and the cursor is in the action of either moving the chart, or resizing the chart

Reviewer: Cant get cursor, to-do on photoshop

You are able to customise your dashboard by adding dashboard filters, controlled filters and additional charts. 

> To resize charts you can simply drag the chart to the size you would like.

Reviewer 2: let's mention they need to hover over the bottom right corner of a chart, i saw people didn't realize that before

  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/example-dashboard-filter-dark.png"
  alt="Hero Dark"
/>
     </Frame>

  </Accordion>
    <Accordion title="4. Filter Preview">

   <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/example-dashboard-filter-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/example-dashboard-filter-dark.png"
  alt="Hero Dark"
/>
     </Frame>

  </Accordion>
</AccordionGroup>

<Tip>For more information on dashboard filters, check out our [Dashboard Filters Guide](/filters/filters-in-depth).</Tip>

#### Add charts

To add additional charts you can go to through the steps of creating a new chart. 

Alternatively if you have already built a chart you can click on **Add Chart** to use on your dashboard.
 <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/add-chart-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/add-chart-dark.png"
  alt="Hero Dark"
/>
     </Frame>

> ## Share your dashboard

Reviewer 2: let's change this to

Share your dashboard and collaborate

> You can share your dashboard via email with anyone in by clicking on the share button and sharing it as follows, if you would like a preview of the dashboard please click on the preview.

Reviewer 2: Missing some words?

with anyone in your organization???

> Your team members will be invited to your dashboard. By sharing your dashboard they will become a team member and can collaborate on future dashboards with you.

Reviewer 2: Rewrite this sentence to:

By sharing your dashboard, they will become a team member of your organization and can view, edit and collaborate on the dashboard(s) with you.

<Frame>
<img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/share.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/share-dark.png"
  alt="Hero Dark"
/>
</Frame>

<Note> If the invitee aceepts the invite, they will gain access to ALL dashboards and data in your organization!</Note>

## User Authorization (Tenant Registration)

To set up **tenant registration** please follow the steps outlined in [Backend Setup](deploy-dashboards/backend-setup). 

Alternatively if you are using our **sample database** and do **not intend to go live** with the dashboard this step is optional.
### Preview dashboard as your designated end users

> If you have completed the [Backend Setup](deploy-dashboards/backend-setup), you are able to view the dashboard as one of your customers.

Reviewer 2: What are the steps to preview it if I completed the BE setup?

> If you are using the **sample database** we have provided two sample tenants, which you are able to preview in the hub below. 

Reviewer 2: Rewrite:

If you do not intend to go live with the dashboard, and thus did not setup your backend accordingly, you could use the platform to create a testing tenant instead and preview the dashboard as the testing tenant. We have provided two sample testing tenants for you for the sample database, which you could use to preview the example dashboard built as the tenant.

 <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/preview-as-dark.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/preview-as-dark.png"
  alt="Hero Dark"
/>
     </Frame>

>      If you are using the **sample database** and want to add more to add more **Preview Tenants**, please click on **Home**, then **Deploy** and then **Registered Tenants**

Reviewer 2: Rewrite to:

If you are using the sample database, the below is the steps and an example of how to create a new testing tenants:

Reviewer 2: also include a link out to the tenancy management page, and ideally a section in that page that explains how tenancy and data permissioning work based on tenancy

     
     Click on **Create tenant**
     You can use the following values to simulate creating a tenant:

- **Tenant Display Name**: `{customer-1}`
- **Filter Column**: `{CustomerID}`
- **Filter Value**: `{User_1}`

> 

Reviewer 2: Add a warning to say:

Any created testing tenant would not be available in live environment. They are only for testing purpose within Upsolve AI's purpose.

 <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/create-tenant.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/create-tenant-dark.png"
  alt="Hero Dark"
/>
 </Frame>

## Deploy your dashboard

Please click on the Deploy button in the Hub.
 <Frame>

  <img
  className="block dark:hidden"
  src="/images/build-your-first-dashboard/light/deploy.png"
  alt="Hero Light"
/>
<img
  className="hidden dark:block"
  src="/images/build-your-first-dashboard/dark/deploy-dark.png"
  alt="Hero Dark"
/>
 </Frame>

> You will need to get the JWT for the current user and the dashboard ID to render the correct dashboard. The JWT is generated by following the steps in Backend Setup. The dashboard ID could be found in the URL when you are viewing the dashboard you want to embed.

Reviewer 2: Link out to BE setup when you mention it
